### DISCUSSION TITLE
Next/Image Cloudinary Transformations

### DISCUSSION THREAD

**Discussion by jacknevitt** [vercel/next.js#34268] [Help]
Is there a way to apply [transformations](https://cloudinary.com/documentation/transformation_reference) to cloudinary urls using next image and the cloudinary loader? I could not see a way to handle these in any of the next image docs.

Using the `quality` prop applies the `q_<quality level>` transformation to the image url. I can't see any other ways to add or change the transformations.

For example, I would like to add a background color to an image (some images are stored with transparency), so that the downloaded file is smaller. I have tried to prefix the path in `src` with `b_rgb:ffffff`, but this just mangles the path.

Or if I need to apply cropping rules to avoid squashing an image with the requested dimensions.

Many thanks!

**Comment by 2stash** [comment]
Yo! I just tried to change the background color and failed.  As you said the documentation does not mention changing some of the things you mention. So I started up a simple project with a png image with transparent background. I tried a few things to get the background color to change and nothing worked, including, using chrome dev tools to modify the wrapping spans that nextjs adds, I tried using ::after and ::before, and of course I tried adding background styles directly to the className assigned to the Image component. 

The only thing that change the image color was the css property 'filter', but that changes the whole image not just the background.

**Comment by jacknevitt** [comment]
Suggestion: Add a `transformations` prop to the Image component that accepts and object with cloudinary transformation values. Merge this object with the default transformations added by Image.
```jsx
<Image {...otherProps} transformations={{ b: 'rgb:ffffff', e: 'art:hokusai' }} />
// appends 'b_rgb:ffffff,e_art:hokusai' to the transformations in the cloudinary url
```
I'm not sure how this prop would work with other image loaders, perhaps they have need for a similar transformations option.
